Assets/Scripts/HandController.cs

```diff
@@ -2,56 +2,66 @@
 using System.Collections.Generic;
 using UnityEngine;
 
-public class HandController : MonoBehaviour
-{
-    public List<GameObject> cards;
-    public List<GameObject> selectedCards;
+public class HandController : MonoBehaviour {
+    public List<CardController> selectedCards;
     public int maxCards;
     public GameObject cardPrefab;
 
     public CastButtonController castButton;
 
-    void Start()
-    {
-        GenerateCards();
+    void Start () {
+        GenerateCards ();
     }
 
-    public void SelectCard(GameObject card) {
-        if (!selectedCards.Contains(card)) {
-            selectedCards.Add(card);
+    public void SelectCard (CardController cardController) {
+        selectedCards.Add (cardController);
+        // Using List.Count to retain the index+1 value for player readability
+        bool canCast = castButton.OnSelectionChanged (selectedCards.Count);
+        if (canCast) {
+            UpdateAllSelectedCards (canCast);
```

Comment:
By not running `UpdateAllSelectedCards()` in the else-case of `SelectCard()`, it means that the CardIndices don't change color when you're adding a card. e.g. If I've selected 3 cards and I select a 4th, the first 3 CardIndices remain red. I assume this is not desired behavior; if I selected 4 cards all of them should become yellow.

However, when you run `DeselectCard()`, you run `UpdateAllSelectedCards()` no matter what. So deselecting will always properly set the colors of the CardIndices.

What this ends up meaning is that `SelectCard()` and `DeselectCard()` should really have identical logic, except for whether they are adding or removing from `selectedCards`. This is why I made them one method. This is what I suggest: 

```c#
    public void HandleCardClick (CardController cardController, bool isSelected) {
        if (isSelected) {
            SelectCard (cardController);
        } else {
            DeselectCard (cardController);
        }
        
        bool canCast = castButton.OnSelectionChanged (selectedCards.Count);

        UpdateAllSelectedCards (canCast);
    }

    public void SelectCard (CardController cardController) {
        selectedCards.Add (cardController);
        // Using List.Count to retain the index+1 value for player readability
    }

    public void DeselectCard (CardController cardController) {
        selectedCards.Remove (cardController);
        cardController.UpdateSelectedCardIndex (0);
    }
```

Reply:
And then in CardController: 

```C#
    void OnMouseUp () {
        isSelected = !isSelected;
        spriteRenderer.color = isSelected ? selectedColor : unselectedColor;

        handController.HandleCardClick(this, isSelected);
    }
```

Reply:
Ah good catch. I agree totally, changed.